packages/next/src/server/lib/awaiter.ts

```diff
@@ -0,0 +1,23 @@
+/**
+ * The Awaiter class is used to manage and await multiple promises.
```

Comment:
This should be expanded a bit. As it reads, `Promise.all` should be sufficient but it actually continues to await if new promises were added in the meantime. Probably also worth noting it can hang indefinitely if an added promise always adds a new one while it hasn't resolved yet.